sway-fmt-v2/src/fmt.rs

```diff
@@ -74,6 +74,10 @@ impl Formatter {
             &mut formatted_code,
             &raw_formatted_code,
         )?;
+        if formatted_code.ends_with('\n') {
+            formatted_code.pop();
```

Comment:
and here

Reply:
Was having some kind of issue with a `None` value returning the block for `Some` when I refactored this as a peekable iterator. May take me a bit to figure out what's going on, this was my short term solution.